src/main/java/org/elasticsearch/ElasticsearchException.java

```diff
@@ -215,4 +192,66 @@ public WithRestHeaders(String msg, Throwable cause, @Nullable ImmutableMap<Strin
             return ImmutableMap.copyOf(map);
         }
     }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.field("type", getExceptionName(this));
+        builder.field("reason", getMessage());
+        innerToXContent(builder, params);
+        return builder;
+    }
+
+    protected void innerToXContent(XContentBuilder builder, Params params) throws IOException {
+        causeToXContent(builder, params);
+    }
+
+    protected void causeToXContent(XContentBuilder builder, Params params) throws IOException {
+        if (getCause() != null) {
+            builder.field("caused_by");
+            builder.startObject();
+            toXContent(builder, params, getCause());
+            builder.endObject();
+        }
+    }
+
+    public static void toXContent(XContentBuilder builder, Params params, Throwable ex) throws IOException {
+        ex = ExceptionsHelper.unwrapCause(ex);
+        if (ex instanceof ElasticsearchException) {
+            ((ElasticsearchException) ex).toXContent(builder, params);
+        } else {
+            builder.field("type", getExceptionName(ex));
+            builder.field("reason", ex.getMessage());
+            if (ex.getCause() != null) {
+                builder.field("caused_by");
+                builder.startObject();
+                toXContent(builder, params, ex.getCause());
+                builder.endObject();
+            }
+        }
+    }
+
+    public Tuple<String, String>[] guessRootCauses() {
+        final Throwable cause = getCause();
+        if (cause != null && cause instanceof ElasticsearchException) {
+            return ((ElasticsearchException) cause).guessRootCauses();
+        }
+        return new Tuple[] { new Tuple<>(getExceptionName(this), getMessage()) };
+    }
+
+    public static Tuple<String, String>[] guessRootCauses(Throwable ex) {
+        ex = ExceptionsHelper.unwrapCause(ex);
+        if (ex instanceof ElasticsearchException) {
+            return ((ElasticsearchException) ex).guessRootCauses();
+        }
+        return new Tuple[] { new Tuple<>(getExceptionName(ex), ex.getMessage()) };
+    }
+
+    protected static String getExceptionName(Throwable ex) {
+        return Strings.toUnderscoreCase(ex.getClass().getSimpleName());
```

Comment:
can we remove an "elasticsearch_" prefix if it exists, I think it will be cleaner? down the road, we can also remove the specific ElasticsearchIllegalArgumentException and such, it was added historically to get the correct status code, but now we also identify IlleglaArgumentException and return the correct status code, so the need for those became irrelevant.